ts-src/bip32.ts

```diff
@@ -362,6 +383,59 @@ export function BIP32Factory(ecc: TinySecp256k1Interface): BIP32API {
         throw new Error('verifySchnorr not supported by ecc library');
       return ecc.verifySchnorr(hash, this.publicKey.subarray(1, 33), signature);
     }
+
+    private tweakFromPublicKey(t: Buffer): Signer {
+      const xOnlyPubKey = toXOnly(this.publicKey);
+      const tweakedPublicKey = ecc.xOnlyPointAddTweak(xOnlyPubKey, t);
+      if (!tweakedPublicKey || tweakedPublicKey.xOnlyPubkey === null)
+        throw new Error('Cannot tweak public key!');
+      const parityByte = Buffer.from([
+        tweakedPublicKey.parity === 0 ? 0x02 : 0x03,
+      ]);
+      const tweakedPublicKeyCompresed = Buffer.concat([
+        parityByte,
+        tweakedPublicKey.xOnlyPubkey,
+      ]);
+
+      return {
+        publicKey: tweakedPublicKeyCompresed,
+        sign: (hash: Buffer, lowR?: boolean): Buffer => {
+          throw new Error('Missing private key');
+        },
+        signSchnorr: (hash: Buffer): Buffer => {
+          throw new Error('Missing private key');
+        },
+        verify: (hash: Buffer, signature: Buffer): boolean => {
+          return ecc.verify(hash, tweakedPublicKeyCompresed, signature);
+        },
+        verifySchnorr: (hash: Buffer, signature: Buffer): boolean => {
+          if (!ecc.verifySchnorr)
+            throw new Error('verifySchnorr not supported by ecc library');
+          return ecc.verifySchnorr(
+            hash,
+            tweakedPublicKey.xOnlyPubkey,
+            signature,
+          );
+        },
+      };
+    }
+
+    private tweakFromPrivateKey(t: Buffer): Signer {
+      const hasOddY =
+        this.publicKey[0] === 3 ||
+        (this.publicKey[0] === 4 && (this.publicKey[64] & 1) === 1);
+      const privateKey = hasOddY
+        ? ecc.privateNegate(this.privateKey!)
+        : this.privateKey;
+
+      const tweakedPrivateKey = ecc.privateAdd(privateKey!, t);
+      if (!tweakedPrivateKey) throw new Error('Invalid tweaked private key!');
+
+      return fromPrivateKey(
```

Comment:
Even though the TS type doesn't have the derive methods etc... The actual returned type does.

We should probably create a thin wrapper that only provides the needed methods and attributes.